src/visualizers/VisualizerGrid.ts

```diff
@@ -0,0 +1,1214 @@
+            this.property4 = Property.Hexagonal_Number
+            this.property4Visualization = PropertyVisualization.Color
+            this.property4MainColor = GREEN
+            this.property5 = Property.Heptagonal_Number
+            this.property5Visualization = PropertyVisualization.Color
+            this.property5MainColor = BLUE
+            this.property6 = Property.Octagonal_Number
+            this.property6Visualization = PropertyVisualization.Color
+            this.property6MainColor = PURPLE
+        }
+    }
+
+    setOverridingSettings() {
+        if (this.pathType === PathType.Rows_Reset_Augmented_By_One) {
+            this.pathType = PathType.Rows
+            this.resetAndAugmentByOne = true
+        }
+
+        if (
+            this.pathType
+            === PathType.Rows_Have_Values_Squared_And_Reset_Augmented_By_One
+        ) {
+            this.formula = Formula.N_SQUARED
+            this.pathType = PathType.Rows
+            this.resetAndAugmentByOne = true
+        }
+
+        if (this.showNumbers && this.amountOfNumbers > 400) {
+            this.amountOfNumbers = 400
+        }
+    }
+
+    setPathVariables(gridSize: number, scalingFactor: number) {
+        //Declare initial movement variables
+        if (this.pathType === PathType.Spiral) {
+            //The starting point is placed so that the whole spiral is centered
+            if (gridSize % 2 === 1) {
+                this.x = (gridSize / 2 - 1) * scalingFactor
+                this.y = (gridSize / 2) * scalingFactor
+            } else {
+                this.x = (gridSize / 2 - 1) * scalingFactor
+                this.y = (gridSize / 2) * scalingFactor
+            }
+        }
+
+        if (this.pathType === PathType.Spiral) {
+            //This uses an increment that decreases every other turn
+            this.numberToTurnAt = 1
+            this.incrementForNumberToTurnAt = 1
+            this.currentDirection = Direction.Right
+            this.whetherIncrementShouldIncrementForSpiral = true
+        } else if (this.pathType === PathType.Rows) {
+            this.currentDirection = Direction.Right
+        }
+    }
+
+    changeDirectionBasedOnPathType(
+        squareRootOfAmountOfNumbers: number,
+        iteration: number
+    ) {
+        //Choose direction for next number
+        if (this.pathType === PathType.Spiral) {
+            //Turn at the numberToTurn at which increases every other turn
+            if (iteration === this.numberToTurnAt) {
+                this.numberToTurnAt += this.incrementForNumberToTurnAt
+                if (this.whetherIncrementShouldIncrementForSpiral) {
+                    this.incrementForNumberToTurnAt += 1
+                    this.whetherIncrementShouldIncrementForSpiral = false
+                } else {
+                    this.whetherIncrementShouldIncrementForSpiral = true
+                }
+
+                if (this.currentDirection === Direction.Right) {
```

Comment:
@oliviaLovesTheSun, I think my question in https://github.com/numberscope/frontscope/pull/164#discussion_r1010866782 was too broad.

In this specific code block, wouldn't it be easier to do `if (this.currentDirection === "right")` etc.? Is there a reason you're using a numeric `enum` for your directions instead of just using strings?

Reply:
I will chime in here that I like enums because they make it clear that there are only/exactly n possibilities (and provide the typo checking that Olivia already mentioned), whereas strings could potentially be very open-ended and leave the reader wondering "hmm, is direction 'southwest' allowed?" -- that sort of thing. There's no ambiguity with what's allowed or not with enums.

Reply:
Ah, the typo checking makes sense now. I was confused because there's a typo in the explanation: "This is also valuable because checking .equals("primes") may have a type while === Preset.Primes can't have a typo."

So we're using an `enum` because:
- it limits the possibilities for directions
- it prevents typos

Is there a clear advantage to using a numeric `enum` to an `enum` with string constants? I'm wondering if perhaps an `enum` like

```
enum Direction {
  Up = "UP",
  Down = "DOWN",
  Left = "LEFT",
  Right = "RIGHT",
}
```

would make debugging easier?

Reply:
Gotcha. I think Olivia wants to stick with the way the code is currently written based on my last conversation with her. Is that correct, @oliviaLovesTheSun?

Reply:
Yes.